homeassistant/components/flux_led/__init__.py

```diff
@@ -19,14 +38,93 @@ async def async_setup(hass: HomeAssistant, config: dict):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Flux LED/MagicLight from a config entry."""
 
-    hass.async_create_task(
-        hass.config_entries.async_forward_entry_setup(entry, "light")
-    )
+    conf = entry.data
+    options = entry.options
+
+    hass.data[DOMAIN][entry.entry_id] = {}
+
+    if options.get(CONF_AUTOMATIC_ADD, conf[CONF_AUTOMATIC_ADD]):
+        scanner_api = BulbScanner()
+        await hass.async_add_executor_job(scanner_api.scan)
+
+        if len(scanner_api.getBulbInfo()) == 0:
+            raise ConfigEntryNotReady("No FluxLED/MagicHome Bulbs found in network.")
+
+        coordinator = FluxLEDListUpdateCoordinator(
+            hass=hass,
+            name="flux_led_scanner",
+            update_interval=DEFAULT_NETWORK_SCAN_INTERVAL,
+            devices=conf[CONF_DEVICES],
+        )
+
+        await coordinator.async_refresh()
+
+        if not coordinator.last_update_success:
+            raise ConfigEntryNotReady
+
+        hass.data[DOMAIN][entry.entry_id] = {BULB_COORDINATOR: coordinator}
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
 
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class FluxLEDListUpdateCoordinator(DataUpdateCoordinator):
```

Comment:
Why do we use a coordinator here? It doesn't seem to have any listeners.

Reply:
The only listener is from the dispatcher that adds new lights if the integration is set to automatically add them.

Reply:
But then we don't need a coordinator for this. We can just have a function that runs on a schedule and uses the dispatcher.

The coordinator is built to fetch data on a schedule and notify more than one listener to get the cached data. If our use case doesn't fit that, we shouldn't use the coordinator. The coordinator won't update on a schedule without attached listeners.

Reply:
Ok - can you provide an example of this type of scheduled function? I can change this.

Reply:
I think I've updated this to just use a @Throttled update properly now.